### PR TITLE
Add assertions on input parameters.

### PR DESCRIPTION
**Pull request by JSAssassin**
addresses #4

### CODE REVIEW

File: assertions.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class Assert {
```
Comment (mattcollier) [inline assertions.js:1]: needs copyright notice.

**Comment by davidlehn** [comment]
I'm a bit late on comments to that assertion code.  Not a big deal in this case, but if this ends up being extracted into something reusable, some thoughts:
- Using a class is not really needed here as there is no state.
- If using a class, could use static methods and export the class itself rather than just a singleton instance.
- To support tree-shaking it would be better to export each method as a function.  (Unless tools are tricky enough to know how to omit unused class methods, but I don't think that happens?)
- Users of that code could then import just the assertion functions they need and webpack or similar would drop out all the other code.  In this case everything is used, but that might not be the case in a general lib.

**Comment by dlongley** [comment]
I'm with @davidlehn here on us not needing a class. Classes should not be used when there's not a reason to -- they are harmful in those cases. If we don't need to track state nor have a need for some inheritance regime, please don't use a class.

**Comment by JSAssassin** [comment]
@dlongley @davidlehn Thank you for the useful feedback. I'll keep that in mind.

**Pull request merged**